dotCMS/src/main/java/com/dotcms/storage/StorageKey.java

```diff
@@ -0,0 +1,113 @@
+package com.dotcms.storage;
+
+/**
+ * Encapsulates a storage key, it has the key (for instance the path)
+ * the group (for instance the bucket, folder, space, etc)
+ * Storage {@link StorageType}
+ * @author jsanca
+ */
+public class StorageKey {
+
+    //Key components
+    private final String path;
+    private final String group;
+    private final StorageType storage;
+
+    /**
+     * Builder based constructor
+     * @param builder
+     */
+    private StorageKey(final Builder builder) {
+
+        this.path = builder.path;
+        this.group = builder.group;
+        this.storage    = builder.storage;
+    }
+
+    /**
+     * read only path
+     * @return
+     */
+    public String getPath() {
+        return path;
+    }
+
+    /**
+     * read only group
+     * @return
+     */
+    public String getGroup() {
+        return group;
+    }
+
+    /**
+     * read-only storage
+     * @return
+     */
+    public StorageType getStorage() {
+        return storage;
+    }
+
+    /**
+     * Convenience builder
+     */
+    public static final class Builder {
+
+        private String path;
+        private String group;
```

Comment:
Codacy found an issue: [Field group has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)